File: BasicMTGScraper/src/main/java/com/wimpy/rest/ScraperRestEndpoint.java
package com.wimpy.rest;


import com.wimpy.core.MtgGoldfishScraper;
import com.wimpy.dao.entity.MtgHistory;
import com.wimpy.model.MtgQuery;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/v1/scrape")
public class ScraperRestEndpoint {

    private static final Logger log = LoggerFactory.getLogger(ScraperRestEndpoint.class);
    private final MtgGoldfishScraper mtgGoldfishScraper;

    @Autowired
    public ScraperRestEndpoint(MtgGoldfishScraper mtgGoldfishScraper) {
        this.mtgGoldfishScraper = mtgGoldfishScraper;
    }


    @GetMapping
    public ResponseEntity<String> getCardDetails(@RequestParam String name, @RequestParam String edition) {
        MtgQuery mtgQuery = new MtgQuery(name, edition, "");

        MtgHistory history = mtgGoldfishScraper.retrieveCardPrice(mtgQuery);

        log.info("name:{}->{}", name, history);

        return ResponseEntity.ok(history.getPrice().toString());
    }

    @PostMapping
    public ResponseEntity<MtgHistory> addNewCard(@RequestParam String name, @RequestParam String link) {
        MtgQuery mtgGoldFishQuery = new MtgQuery(name, "", link);


        return ResponseEntity.ok(mtgGoldfishScraper.saveCardPrice(mtgGoldFishQuery));
    }
}

File: BasicMTGScraper/src/main/kotlin/com/wimpy/rest/v1/ScraperRestEndpoint.kt
package com.wimpy.rest.v1

import com.wimpy.core.MtgGoldfishScraper
import com.wimpy.db.dao.MtgHistory
import com.wimpy.rest.v1.model.MtgQuery
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("/v1/scrape")
class ScraperRestEndpoint @Autowired constructor(private val mtgGoldfishScraper: MtgGoldfishScraper) {

    private val log = LoggerFactory.getLogger(this.javaClass)


    @PostMapping
    fun addNewCard(
        @RequestParam(required = false, defaultValue = "") name: String,
        @RequestParam link: String
    ): ResponseEntity<List<MtgHistory>> {
        val mtgHistory = mtgGoldfishScraper.retrieveCardPrice(MtgQuery(name, "", link))
        return ResponseEntity.ok(mtgGoldfishScraper.saveCardPrice(mtgHistory))
    }


}